habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -254,6 +264,20 @@ def get_local_surface_center(
         local_center.y = self.bounds.y().min
         return local_center
 
+    def check_if_point_on_surface(
+        self,
+        sim: habitat_sim.Simulator,
+        point: mn.Vector3,
+        threshold: float = 0.05,
+    ) -> bool:
+        local_point = self.get_local_transform(sim).transform_point(point)
+        bounds = self.bounds
+        return (
+            bounds.x().contains(local_point.x)
+            and bounds.z().contains(local_point.z)
+            and np.abs(bounds.y().min - local_point.y) < threshold
```

Comment:
Do we consider y as the up direction?

Reply:
Thanks for catching this. Yes, y is the up direction but I need to use `self.up_axis`. Will fix this.

Reply:
Updated